sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,532 @@
+        )
+        self._transformer_backbone = (
+            self._config["transformer_backbone"]
+            if "transformer_backbone" in self._config.keys()
+            else self.transformer_backbone
+        )
+        self._criterion = (
+            self._config["criterion"]
+            if "criterion" in self._config.keys()
+            else self.criterion
+        )
+        # evaluate the sequence length passed by the user
+        self._seq_len = (
+            self._config["seq_len"]
+            if "seq_len" in self._config.keys()
+            else self.seq_len
+        )
+        if self._seq_len > self._moment_seq_len:
+            warnings.warn(
+                f"length of {self._seq_len} was found which is greater than 512. "
+                "The most recent 512"
+                " values will be used when fitting.",
+                stacklevel=2,
+            )
+            self._seq_len = 512
+        # in the case the config contains 'forecasting_horizon', we'll set
+        # fh as that, otherwise we override it using the fh param
+        self._fh_config = (
+            self._config["forecasting_horizon"]
+            if "forecasting_horizon" in self._config.keys()
+            else self.fh
+        )
+        if fh is not None:
+            self._fh_input = max(fh.to_relative(self.cutoff))
+        self._fh = self._fh_input if fh is not None else self._fh_config
+        train_val_split = int(len(y) * (1 - self.train_val_split))
+        self._model = momentfm.MOMENTPipeline.from_pretrained(
+            self._pretrained_model_name_or_path,
+            model_kwargs={
+                "task_name": "forecasting",
+                "dropout": self._dropout,
+                "head_dropout": self._head_dropout,
+                "freeze_encoder": self._freeze_encoder,
+                "freeze_embedder": self._freeze_embedder,
+                "seq_len": 512,  # forced to be hard coded
+                "freeze_head": self._freeze_head,
+                "device": self._device,
+                "transformer_backbone": self._transformer_backbone,
+                "forecast_horizon": self._fh,
+            },
+        )
+        self._model.init()
+        self._y_cols = y.columns
+        self._y_shape = y.values.shape
+        train_dataset = momentPytorchDataset(
+            y=y[:train_val_split],
+            fh=self._fh,
+            seq_len=self._seq_len,
+        )
+
+        train_dataloader = DataLoader(
+            train_dataset, batch_size=self.batch_size, shuffle=True
+        )
+
+        val_dataset = momentPytorchDataset(
+            y=y[train_val_split:],
+            fh=self._fh,
+            seq_len=self._seq_len,
+        )
+
+        val_dataloader = DataLoader(
+            val_dataset, batch_size=self.batch_size, shuffle=True
+        )
+
+        criterion = self._criterion
+        optimizer = Adam(self._model.parameters(), lr=1e-4)
```

Comment:
This is training all parameters of the model. Is this the proposed way of the authors? I would think that only the heads should be retrained

Reply:
This is the proposed way of training by the authors. In the model specification, we specify which components of the model we would like to tune on via

```
            model_kwargs={
                "task_name": "forecasting",
                "dropout": self._dropout,
                "head_dropout": self._head_dropout,
                "freeze_encoder": self._freeze_encoder,
                "freeze_embedder": self._freeze_embedder,
                "seq_len": 512,  # forced to be hard coded
                "freeze_head": self._freeze_head,
                "device": self._device,
                "transformer_backbone": self._transformer_backbone,
                "forecast_horizon": self._fh,
            },
```
By default, we set `freeze_encoder` and `freeze_embedder` to be equal to `True`, so those parameters will be frozen, so they won't be updated when we initialize the optim through `Adam`. 

If the user specifies those two parameters to be `False`, then yes you are correct and it will re-train all of the weights in the model.

Reply:
By default `freeze_head` is set to `False`